pkg/crc/preflight/preflight_checks_linux.go

```diff
@@ -366,6 +366,12 @@ func removeLibvirtCrcNetwork() error {
 
 func removeCrcVM() error {
 	logging.Debug("Removing 'crc' VM")
+
+	if err := os.RemoveAll(constants.MachineInstanceDir); err != nil {
```

Comment:
One key thing to know is that `os.RemoveAll("/non/existing/path")` will return `nil` if any of the path components is non-existing. Without this, this would have fixed one corner case, and broken another one.
I'm mentioning this for posterity as I had to check how `os.RemoveAll` worked :)